src/core/actors/Tribe.java

```diff
@@ -230,117 +205,89 @@ public Vector2d getPosition()
         return null;
     }
 
-
-    public boolean isMonumentBuildable(Types.BUILDING building)
-    {
-        return monuments.get(building) == MONUMENT_STATUS.AVAILABLE;
+    public ArrayList<Types.TRIBE> getTribesMet(){
+        return tribesMet;
     }
 
-    public void monumentIsBuilt(Types.BUILDING building)
-    {
-        monuments.put(building, MONUMENT_STATUS.BUILT);
-    }
+    public void meetTribe(GameState gs, int tribeID){
 
-    public int getnumKills() {
-        return nKills;
-    }
+        Board b = gs.getBoard();
+        Tribe[] t = gs.getTribes(); // get tribes from boards
 
-    public void addKill() {
-        this.nKills++;
+        boolean[] inMetTribes = new boolean[4];
+        //loop through all tribes
+        for(int i = 0; i<t.length; i++){
+            // if tribes not in tribes met or tribe is itself then do nothing else add to tribesmet arraylist
+            if (t[i].tribe == this.tribesMet.get(i) || t[i].tribeId == tribeID){
+                inMetTribes[i] = true;
+            }
+            if(!inMetTribes[i]){
+                tribesMet.add(t[i].tribe); // add to this tribe
+                t[i].tribesMet.add(this.tribe); // add to met tribe as well
+
+                //Pick a technology at random from the tribe to learn
+                TechnologyTree thisTribeTree = getTechTree();
+                TechnologyTree metTribeTree = t[i].getTechTree();
+                ArrayList<Types.TECHNOLOGY> techInThisTribe = new ArrayList<>(); //Check which tech in this tribe
+                ArrayList<Types.TECHNOLOGY> techInMetTribe = new ArrayList<>(); // Check which tech in met tribe
+                //Check which technologies both research trees contain
+                Random r = new Random();
```

Comment:
We don't create new Random() objects, so we can keep all randomness down to one single seed. The only Random object is in GameState. We may consider having a Singleton to avoid this.